internal/collector/recon/sharding.go

```diff
@@ -0,0 +1,608 @@
+		containerShardingVisitedSuccess: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_visited_success",
+				Help: "Container shard visited number of successes reported by the swift-recon tool.",
+			}, []string{"storage_ip"}),
+		containerShardingInProgressActive: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_active",
+				Help: "Container sharding in progress number of shards active reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingInProgressCleaved: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_cleaved",
+				Help: "Container sharding in progress number of shards cleaved reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingInProgressCreated: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_created",
+				Help: "Container sharding in progress number of shards created reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingInProgressError: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_error",
+				Help: "Container sharding in progress number of errors reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingInProgressFound: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_found",
+				Help: "Container sharding in progress number found reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingInProgressObjectcount: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_in_progress_object_count",
+				Help: "Container sharding in progress object count reported by the swift-recon tool.",
+			}, []string{"storage_ip", "container", "account"}),
+		containerShardingCandidatesFound: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_candidates_found",
+				Help: "Number of container sharding candidates reported by the swift-recon tool.",
+			}, []string{"storage_ip"}),
+		containerShardingCandidatesObjectCount: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: "swift_cluster_containers_sharding_candidates_object_count",
+				Help: "Container sharding candidates object count reported by the swift-recon tool.",
+			}, []string{"storage_ip", "account", "container"}),
+	}
+}
+
+// Name implements the collector.Task interface.
+func (t *ShardingTask) Name() string {
+	return "recon-sharding"
+}
+
+// DescribeMetrics implements the collector.Task interface.
+func (t *ShardingTask) DescribeMetrics(ch chan<- *prometheus.Desc) {
```

Comment:
Please add a `//nolint:dupl` line before this one to silence the linter. Same on the CollectMetrics function.

Reply:
Done